flake.nix

```diff
@@ -130,64 +131,51 @@
 
           modules = haskellModules system;
         };
-    in
-    {
+
       project = perSystem projectFor;
       flake = perSystem (system: (projectFor system).flake { });
+    in
+    flake-parts.lib.mkFlake { inherit inputs; } {
+      imports = [
+        ./pre-commit.nix
+      ];
+      flake = {
+        inherit project flake;
+
+        # This will build all of the project's executables and the tests
+        check = perSystem
+          (system:
+            (nixpkgsFor system).runCommand "combined-check"
+              {
+                nativeBuildInputs = builtins.attrValues self.checks.${system}
+                  ++ builtins.attrValues self.flake.${system}.packages;
+              } ''mkdir $out''
+          );
```

Comment:
We don't do this any longer, Hercules CI default just checks everything. Nicer UX as well